app/src/main/java/com/android/ubclaunchpad/driver/UI/DestinationActivity.java

```diff
@@ -94,6 +94,45 @@ public void onClick(View v) {
             }
         });
 
+        currentAutoCompleteFragment.setOnPlaceSelectedListener(new PlaceSelectionListener() {
+            @Override
+            public void onPlaceSelected(Place place) {
+                Log.d(TAG, "Place: " + place.getName() + "\nLatLong: " + place.getLatLng());
+
+                okButton.setEnabled(true);
+
+                User innerUser;
+
+                try {
+                    innerUser = UserManager.getInstance().getUser();
+                    if (innerUser != null) {
+                        // Save location to user's currentLatLngStr
+                        innerUser.setCurrentLatLngStr(place.getLatLng());
+                    } else {
+                        FirebaseUtils.getFirebaseAuth().signOut();
+                        startActivity(new Intent(DestinationActivity.this, LoginActivity.class));
+                        finish();
+                    }
+                } catch (Exception e) {
+                    Log.e(TAG, "Could not retrieve user" + e.getMessage());
+                }
+
+                if (FirebaseUtils.getFirebaseUser() != null) {
+                    String uid = FirebaseUtils.getFirebaseUser().getUid();
+                    Log.d(TAG, "got uid: " + uid);
+                    mDatabase.child(StringUtils.FirebaseUserEndpoint)
+                            .child(uid)
+                            .child(StringUtils.FirebaseDestinationLatLngEndpoint)
```

Comment:
So if you use StringUtils.FirebaseDstinationLatLngEndpoint that is updating the destinationLatLngStr in Firebase so u should change this to update currentLatLngStr